packages/flame_studio/lib/src/core/game_controller.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flame/game.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final gameControllerProvider =
+    StateNotifierProvider<_GameController, _GameState>(
+  (ref) => _GameController(),
+);
+
+class _GameState {
```

Comment:
I am not sure how riverpod works, but should this class implement equals?

Reply:
Good catch, it should indeed. It will still work most of the time now though since it always creates a new object, but if the new object would have the same data as the old object it would re-render unnecessarily.

Reply:
Let's pull it in when we have a few more places that need it.
It's also very nice to stay away from code generation until it's needed. 😅

Reply:
Added `operator==`